File: PayTR Direkt API/PayTR Taksit Oranları Sorgulama Servisi/English/SAMPLE .NET/Installment_sample.cs
using Newtonsoft.Json.Linq; 
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Net;
using System.Security.Cryptography;
using System.Text;
using System.Web;
using System.Web.Script.Serialization;
using System.Web.UI;
using System.Web.UI.WebControls;

public partial class taksit_ornek : System.Web.UI.Page
{
    protected void Page_Load(object sender, EventArgs e) {

        // ########################################
        //
        string merchant_id      = "XXXXXX";
        string merchant_key     = "YYYYYYYYYYYYYY";
        string merchant_salt    = "ZZZZZZZZZZZZZZ";
        //
        string request_id       = "";
        //
        // ########################################
        NameValueCollection data = new NameValueCollection();
        data["merchant_id"] = merchant_id;
        data["request_id"] = request_id;
      
        // ######################################################################
        string Birlestir = string.Concat(merchant_id, request_id, merchant_salt);
        HMACSHA256 hmac = new HMACSHA256(Encoding.UTF8.GetBytes(merchant_key));
        byte[] b = hmac.ComputeHash(Encoding.UTF8.GetBytes(Birlestir));
        data["paytr_token"] = Convert.ToBase64String(b);
        // ######################################################################

        using (WebClient client = new WebClient()) {
            client.Headers.Add("Content-Type", "application/x-www-form-urlencoded");
            byte[] result = client.UploadValues("https://www.paytr.com/odeme/taksit-oranlari", "POST", data);
            string ResultAuthTicket = Encoding.UTF8.GetString(result);
            dynamic json = JValue.Parse(ResultAuthTicket);

            if (json.status == "success") {
       
                Response.Write(json);
            }else{ 

                Response.Write(json.err_msg);
            }
        }
    }
}

File: PayTR Direkt API/PayTR Taksit Oranları Sorgulama Servisi/English/Sample PYTHON/Installment_sample.py
# Python 3.6+

import base64
import hmac
import hashlib
import requests
import json
import time


merchant_id = 'XXX'
merchant_key = b'XXX'
merchant_salt = 'XXX'


request_id = str(time.time())


hash_str = merchant_id + request_id + merchant_salt
paytr_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode(), hashlib.sha256).digest())

params = {
    'merchant_id': merchant_id,
    'request_id': request_id,
    'paytr_token': paytr_token
}

result = requests.post('https://www.paytr.com/odeme/taksit-oranlari', params)
res = json.loads(result.text)

if res['status'] == 'success':

    print(res)
else:
    """
    Örn.
    ['status']        - error
    ['err_no']        - 006
    ['err_msg']       - Zorunlu alan degeri gecersiz veya gonderilmedi: 
    """
    print(res)